### PR TITLE
[trace][bugfix] Always get latest tracer

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Always get latest tracer, so that it can get the latest resource attributes.

As shown in below screenshot, when start trace with new collection name "xyz", the collection value in resource attributes does not change until we get a new tracer:

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by thy09 — DISMISSED** [review]

File: src/promptflow-core/promptflow/executor/flow_executor.py [hunk 1]
```diff
@@ -62,12 +62,7 @@
 from promptflow.tracing._integrations._openai_injector import inject_openai_api
 from promptflow.tracing._operation_context import OperationContext
```
Comment (thy09) [inline src/promptflow-core/promptflow/executor/flow_executor.py:63]: Maybe add a test case for the scenario mentioned in the description?

Comment (zhengfeiwang) [reply src/promptflow-core/promptflow/executor/flow_executor.py]: updated the fix way, and add a test in start trace UT, may take a look

**Review by thy09 — APPROVED** [review]

File: src/promptflow-tracing/tests/e2etests/test_tracing.py [hunk 2]
```diff
@@ -222,6 +223,22 @@ def assert_otel_traces_with_embedding(self, dev_connections, func, inputs, expec
     def test_otel_trace_with_multiple_functions(self):
         execute_function_in_subprocess(self.assert_otel_traces_with_multiple_functions)
 
+    def test_otel_tracer_refreshed_after_start_trace(self):
+        from promptflow.tracing import start_trace
+
+        memory_exporter = prepare_memory_exporter()
```
Comment (thy09) [inline src/promptflow-tracing/tests/e2etests/test_tracing.py:229]: Should we use subprocess to run the test to avoid side effect?

**Review by zzn2 — DISMISSED** [review]

File: src/promptflow-tracing/tests/e2etests/test_tracing.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow-tracing/tests/e2etests/test_tracing.py:229]: do other test use subprocess? i just copy lines, adding `start_trace`.

**Review by zzn2 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

File: src/promptflow-tracing/tests/e2etests/test_tracing.py (hunk 2 above)
Comment (thy09) [reply src/promptflow-tracing/tests/e2etests/test_tracing.py:229]: You can see the test_tracing, which is using a function execute_function_in_subprocess

Comment (zhengfeiwang) [reply src/promptflow-tracing/tests/e2etests/test_tracing.py:229]: I see, let me update this

**Review by zzn2 — APPROVED** [review]

**Pull request merged**